### PR TITLE
Upgrade to Rocket v0.5

### LINKED ISSUE #30: Rocket 0.5
Now that Rocket 0.5-rc is out, it would be great to see this updated to Rocket 0.5.

### PR DESCRIPTION
**Pull request by aburd**
Closes https://github.com/TatriX/realworld-rust-rocket/issues/30

### CODE REVIEW

File: src/database/mod.rs [hunk 1]
```diff
@@ -1,17 +1,15 @@
-use rocket_contrib::databases::diesel;
-
 pub mod articles;
 pub mod comments;
 pub mod profiles;
 pub mod users;
 
 #[database("diesel_postgres_pool")]
-pub struct Conn(diesel::PgConnection);
+pub struct Db(diesel::PgConnection);
```
Comment (aburd) [inline src/database/mod.rs:7]: https://github.com/SergioBenitez/Rocket/blob/master/examples/databases/src/diesel_sqlite.rs

Comment (aburd) [reply src/database/mod.rs:7]: I changed the naming based off one of the official examples. I figured people would probably be referring back to that at some point.

File: src/routes/articles.rs [hunk 2]
```diff
@@ -38,61 +36,79 @@ pub fn post_articles(
...
 
-    let article = db::articles::create(
-        &conn,
-        auth.id,
-        &title,
-        &description,
-        &body,
-        &new_article.tag_list,
-    );
+    let article = db
```
Comment (aburd) [inline src/routes/articles.rs:39]: This is an example of why I renamed the `db` module to `database`.
To differentiate between the module and the wrapped connection coming into the function.

File: tests/common.rs [hunk 3]
```diff
@@ -22,14 +23,12 @@ macro_rules! json_string {
...
 
-
-pub fn test_client() -> &'static Client {
-    static INSTANCE: OnceCell<Client> = OnceCell::new();
+pub fn test_client() -> &'static Mutex<Client> {
+    static INSTANCE: OnceCell<Mutex<Client>> = OnceCell::new();
     INSTANCE.get_or_init(|| {
         let rocket = realworld::rocket();
-        Client::new(rocket).expect("valid rocket instance")
+        Mutex::from(Client::tracked(rocket).expect("valid rocket instance"))
```
Comment (aburd) [inline tests/common.rs:30]: blocking::Client now has a negative trait implementation for Send

**Comment by TatriX** [comment]
Thank you for the contribution!
I will try to do a review and test the code sometime this week.

**Comment by aburd** [comment]
@TatriX 
Much appreciated. I am sure there are a lot of places that can be improved.
Thanks again for your time.

**Pull request merged**